### ISSUE TITLE
Consistency in command line interface

### ISSUE THREAD

**Issue by anurse** [dotnet/cli#1476]
There are some inconsistencies in the command line interface for dotnet-cli that need to be resolved. This issue is to help track those (for larger items or items in other repos, create a new issue and link it here) /cc @piotrpMSFT @davidfowl @joelverhagen
- [x] NuGet console output is prefixed with `info:`, etc. and looks inconsistent. The consistent prefixing is kind of nice, but might be too verbose for other CLI things. Maybe we just remove it from NuGet. We should still ensure errors/warnings in NuGet are easily `grep`able
- [x] Colors are not properly forwarded when dotnet-cli shells out. We can just remove the redirection-by-default in Command.cs to fix this AFAIK. It only becomes an issue when we want to forward stdout/err AND capture it, which we don't ever do right now. /cc @brthor - we talked about this last week
- [ ] Should we try and put `AnsiConsole` somewhere more common? It's very useful for an application to be able to rely on always using ANSI. Especially since TH2 Win10 builds support ANSI in `conhost` (and `AnsiConsole` could detect this and avoid doing the ANSI-emulation code we do). Maybe this belongs in `System.CommandLine`?
- [x] `dotnet restore`'s help text is weird (see below). It's mainly because the "app name" is defined in `NuGet.CommandLine.XPlat.dll`, so we made it `nuget3` when it was originally defined. We could just make it `dotnet restore` but it seems weird to have the XPlat nuget command line know about `dotnet`... This also goes to figuring out if there is going to be a `nuget restore` cross-plat, and what that would be.

There are probably more. Please feel free to add items directly to the list, as well as attaching bugs to items and marking items as completed when they are resolved!.

`dotnet restore` help text:

```
ps1 # dotnet help restore
Usage: nuget3 restore [arguments] [options]
Arguments:
  [root]  List of projects and project folders to restore. Each value can be: a path to a project.json or global.json file, or a folder to recursively search for project.json files.
Options:
  -h|--help                       Show help information
  -s|--source <source>            Specifies a NuGet package source to use during the restore.
  --packages <packagesDirectory>  Directory to install packages in.
  --disable-parallel              Disables restoring multiple projects in parallel.
  -f|--fallbacksource <FEED>      A list of packages sources to use as a fallback.
  --runtime <RID>                 List of runtime identifiers to restore for.
  --configfile <file>             The NuGet configuration file to use.
  -v|--verbosity <verbosity>      The verbosity of logging to use. Allowed values: Debug, Verbose, Information, Minimal, Warning, Error.
```

**Comment by bricelam** [comment]
I wish verbosity was handled more consistently and globally. We currently have awkward things like the following.

    dotnet --verbose restore --verbosity verbose

**Comment by brthor** [comment]
@anurse I think @AustinWise PR #1212 fixed the color forwarding issue (On Windows at least)

That being said, not doing the line buffered forwarding by default does seem right. The only place we capture output is for `dotnet-compile-csc` and in tests.

**Comment by blackdwarf** [comment]
In general, we have covered most of the things in this issue. If something crops up that is not covered, please open a new issue. Thanks!
